pandas/core/indexes/multi.py

```diff
@@ -1474,6 +1553,30 @@ def _sort_levels_monotonic(self):
                           names=self.names, sortorder=self.sortorder,
                           verify_integrity=False)
 
+    def squeeze(self):
+        """
+        Squeeze a single level MultiIndex to be a regular Index instance.
+
```

Comment:
I don't think we need to make this a public method.

Reply:
The rationale for making this a public method was, in bigger projects I will sometimes receive dataframes that I have no control over. Sometimes the columns would be formatted as a single level multiindex (I think this was mostly due to processes setup in earlier versions of pandas where the index would otherwise have been converted automatically to an Index). The easiest way to deal with these cases was to "squeeze" the index.
It's not a common occurrence though. Is this too obscure to justify keeping it public?

Also the context that it is used here is in a classmethod, so it's being called on the MultiIndex object externally. Would it be ok to call a "private" method in this fashion?

Reply:
changed squeeze -> _squeeze

Reply:
I am re-thinking if this should be public, see #22866